kubectl-volsync/cmd/migration.go

```diff
@@ -19,12 +19,86 @@ package cmd
 import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/kubectl/pkg/util/i18n"
 	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	volsyncv1alpha1 "github.com/backube/volsync/api/v1alpha1"
 )
 
 // MigrationRelationship defines the "type" of migration Relationships
-const MigrationRelationship RelationshipType = "migration"
+const MigrationRelationshipType RelationshipType = "migration"
+
+// migrationRelationship holds the config state for migration-type
+// relationships
+type migrationRelationship struct {
+	Relationship
+	data migrationRelationshipData
+}
+
+// migrationRelationshipData is the state that will be saved to the
+// relationship config file
+type migrationRelationshipData struct {
+	Version     int
+	Source      *migrationRelationshipSource
+	Destination *migrationRelationshipDestination
+}
+
+type migrationRelationshipSource struct {
+	// Volume to be migrated
+	Volume string
+	// Total volume size
+	Size *resource.Quantity
+}
```

Comment:
The external -> kubernetes workflow doesn't really have a "source" for the relationship.
The Size parameter can be stored in Destination.Capacity

Reply:
Shall we emphasize the capacity description as it is the size of the volume to be migrated? something like

`migrationCreateCmd.Flags().String("capacity", "", "Size of the standalone volume to be migrated. The PVC that is to be created is of this size. format ex: 10Gi")`

Reply:
I'd like to keep it clear that the capacity is the PVC size. While it's true that the copy will fail if the PVC is too small, it needs to be really obvious that's what they are setting w/ this option. There are valid reasons to both oversize (for growth) or undersize (not migrating everything).

Reply:
makes sense